xarray/core/datatree_io.py

```diff
@@ -74,10 +89,12 @@ def _datatree_to_netcdf(
                 unlimited_dims=unlimited_dims.get(node.path),
                 **kwargs,
             )
-        mode = "r+"
+        mode = "a"
```

Comment:
I am 99% sure that the modes "a" and "r+" are equivalent. I swapped to "a" in order to comply with the `NetCdfWriteModes` typing, which was `["a", "w"]` everywhere else.

Reply:
I think I agree that that.